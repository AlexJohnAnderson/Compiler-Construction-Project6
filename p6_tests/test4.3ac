[BEGIN GLOBALS]
main
[END GLOBALS]
[BEGIN main LOCALS]
[END main LOCALS]
main:       enter main
lbl_1:      nop
            goto lbl_1
lbl_2:      nop
lbl_0:      leave main

