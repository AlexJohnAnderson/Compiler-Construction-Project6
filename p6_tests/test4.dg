int main(){
    for(int i; i<3; i++)
    {
        output(i);
    }
}