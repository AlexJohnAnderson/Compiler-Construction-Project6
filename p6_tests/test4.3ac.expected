OUTPUT FILE
[BEGIN GLOBALS]
main
[END GLOBALS]
[BEGIN main LOCALS]
i (local var of 8 bytes)
tmp0 (tmp var of 8 bytes)
[END main LOCALS]
main:       enter main
lbl_1:      nop
            [tmp0] := [i] LT64 3
            IFZ [tmp0] GOTO lbl_2
            REPORT [i]
            [i] := [i] ADD64 1
            goto lbl_1
lbl_2:      nop
lbl_0:      leave main
